games-emulation/atari++/patches/atari++-1.81.patchset

```diff
@@ -0,0 +1,52 @@
+From 2f81843d221f595388fb3853da149e5bc2ee17e4 Mon Sep 17 00:00:00 2001
+From: Augustin Cavalier <waddlesplash@gmail.com>
+Date: Sun, 27 May 2018 10:24:08 +0200
+Subject: [PATCH] Fix autoconf script to include the proper header for timeval.
+
+
+diff --git a/configure.in b/configure.in
+index 636ba18..846ea1c 100644
+--- a/configure.in
++++ b/configure.in
+@@ -163,7 +163,7 @@ AC_CHECK_FUNCS([strcasecmp strncasecmp strchr strerror strrchr strtol strtod])
+ AC_CHECK_FUNCS([time localtime snprintf vsnprintf usleep select getenv isatty chmod])
+ AC_CHECK_FUNCS([tcsetattr tcgetattr tcflush tcdrain cfmakeraw cfsetospeed cfsetispeed])
+ AC_CHECK_FUNCS([ioctl open creat close read write unlink fileno])
+-AC_CHECK_TYPE([struct timeval],[AC_DEFINE(HAS_STRUCT_TIMEVAL,[1],[Define to 1 if struct timeval is available])])
++AC_CHECK_TYPE([struct timeval],[AC_DEFINE(HAS_STRUCT_TIMEVAL,[1],[Define to 1 if struct timeval is available])],[],[[#include <sys/time.h>]])
+ #
+ #
+ # Check for where X11 could be
+-- 
+2.16.2
+
+
+From f5c9c8cea26f61c5b955e788246695ae141c1465 Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Sun, 27 May 2018 10:25:04 +0200
+Subject: don't use ecvt_r
+
+
+diff --git a/mathpackpatch.cpp b/mathpackpatch.cpp
+index c8c8e1e..9ab2695 100644
+--- a/mathpackpatch.cpp
++++ b/mathpackpatch.cpp
+@@ -300,6 +300,7 @@ void MathPackPatch::FASC(class AdrSpace *adr,class CPU *cpu)
+   // naturally an exponential notation one digit sooner than the C
+   // standard.
+ #if HAVE_ECVT_R
++#ifndef __HAIKU__
```

Comment:
May I ask why `ecvt_r` is disabled?

Reply:
Iirc I got an error on it not being defined, switching to "ecvt" gave me another error ...

Reply:
`HAVE_ECVT_R` should only be defined when `configure` found `ecvt_r` in the system. I wonder why it's defined if it can't be used 😕

Reply:
Been looking around a bit, seems stdlib.h provides ecvt but not ecvt_r ... config.log doesn't really provide any answer to where it found it ...